src/spreadinterp.cpp

```diff
@@ -598,11 +591,10 @@ static void interp_square(T *FINUFFT_RESTRICT target, const T *du, const T *ker1
   // no wrapping: avoid ptrs
   using arch_t                          = typename simd_type::arch_type;
   static constexpr auto padding         = get_padding<T, 2 * ns>();
-  static constexpr auto alignment       = arch_t::alignment();
   static constexpr auto simd_size       = simd_type::size;
   static constexpr uint8_t line_vectors = (2 * ns + padding) / simd_size;
-  if (i1 >= 0 && i1 + ns <= N1 && i2 >= 0 && i2 + ns <= N2 &&
-      (i1 + ns + (padding + 1) / 2 < N1)) {
+  if (i1 >= 0 && i1 + ns <= BIGINT(N1) && i2 >= 0 && i2 + ns <= BIGINT(N2) &&
```

Comment:
Type juggling to avoid warnings. In such a situation it is usually safer to cast the unsigned type to signed, because otherwise negative numbers could be cast to very large positive ones, screwing up the comparison.

Reply:
we could also disable the unsigned compare entirely. Any dramatic error that can arise from that?

Reply:
You mean disable the warning? I think this can indeed cause dramatic errors.
If one of the arguments in a comparison is unsigned, the the other will be converted to unsigned as well.
This leads to the absurd case that `unsigned(1)<-1` is `true`. So I think this warning is quite valuable.

Reply:
I see, thanks for fixing.